soroban-sdk/src/prng.rs

```diff
@@ -0,0 +1,48 @@
+//! Prng contains functions for pseudo-random functions.
+use crate::{
+    env::internal, unwrap::UnwrapInfallible, Bytes, Env, IntoVal, TryIntoVal, Val,
+    Vec,
+};
+
+/// Prng provides access to pseudo-random  functions.
+pub struct Prng {
+    env: Env,
+}
+
+impl Prng {
+    pub(crate) fn new(env: &Env) -> Prng {
+        Prng { env: env.clone() }
+    }
+
+    pub fn env(&self) -> &Env {
+        &self.env
+    }
+
+    // Reseeds the pseudorandom number generator (PRNG) with the provided `seed` value.
+    pub fn prng_reseed(&self, seed: &Bytes) {
+        let env = self.env();
+        internal::Env::prng_reseed(env, seed.into()).unwrap_infallible();
+    }
+
+    // Returns a random u64 in the range between `lower` and `upper` inclusive.
+    pub fn prng_u64_in_inclusive_range(&self, lower: u64, upper: u64) -> u64 {
+        let env = self.env();
+        internal::Env::prng_u64_in_inclusive_range(env, lower.into(), upper.into())
+            .unwrap_infallible()
+            .into()
+    }
+
+    // Shuffles a given vector v using the Fisher-Yates algorithm.
+    pub fn prng_vec_shuffle<V>(&self, v: V) -> Vec<Val>
```

Comment:
Could we remove `vec` from the function name? This will be usable with non-Vec's because it internally takes care of converting things to a Vec to make it work. The vec feels unnecessarily restrictive in name when not in implementation.
```suggestion
    pub fn shuffle<V>(&self, v: V) -> Vec<Val>
```